Source/WebCore/dom/ExtensionStyleSheets.cpp

```diff
@@ -90,15 +90,15 @@ void ExtensionStyleSheets::clearPageUserSheet()
 {
     if (m_pageUserSheet) {
         m_pageUserSheet = nullptr;
-        m_document.styleScope().didChangeStyleSheetEnvironment();
+        m_document->styleScope().didChangeStyleSheetEnvironment();
```

Comment:
How come we're not protecting the document here? Also, it'd be good to call the existing `checkedStyleScope()`.

Reply:
It's because `styleScope()` is a trivial function but you're right that we should be using `checkedStyleScope()` here.

Reply:
styleScope() is a trivial function but we're calling didChangeStyleSheetEnvironment() on it, which is not trivial. This is why I would protect the document, in order to protect the style scope (which is not ref counted), and then get a checkedStyleScope() from the document.

Reply:
Hm... okay. I guess we could do that.